### PR TITLE
Move solo8 -> solo8ti and solo8 with micro drivers.

### PR DESCRIPTION
**Pull request by jviereck**

### CODE REVIEW

File: demos/demo_solo8.cpp [hunk 1]
```diff
@@ -51,10 +51,11 @@ static THREAD_FUNCTION_RETURN_TYPE control_loop(void* robot_void_ptr)
                 sliders_filt_buffer[i].pop_front();
             }
             sliders_filt_buffer[i].push_back(sliders(i));
-            sliders_filt(i) = std::accumulate(sliders_filt_buffer[i].begin(),
-                                              sliders_filt_buffer[i].end(),
-                                              0.0) /
-                              (double)sliders_filt_buffer[i].size();
+            // sliders_filt(i) = std::accumulate(sliders_filt_buffer[i].begin(),
```
Comment (MaximilienNaveau) [inline demos/demo_solo8.cpp]: to be removed

File: include/blmc_robots/solo8.hpp [hunk 2]
```diff
@@ -1,14 +1,15 @@
 /**
- * \file quadruped.hpp
- * \author Manuel Wuthrich
- * \date 2018
- * \copyright Copyright (c) 2019, New York University and Max Planck Gesellschaft.
+ * \file solo8.hpp
```
Comment (MaximilienNaveau) [inline include/blmc_robots/solo8.hpp]: remove the file name. just @file is enough

File: include/blmc_robots/solo8.hpp [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+ * \author Julian Viereck
```
Comment (MaximilienNaveau) [inline include/blmc_robots/solo8.hpp]: remove @date + @ author.
Add @brief and a short description of the file.

File: include/blmc_robots/solo8.hpp [hunk 4]
```diff
@@ -306,28 +307,31 @@ class Solo8
    */
   Eigen::Vector4d contact_sensors_states_;
 
+  /** @brief Map the joint id to the motor board id, @see Solo12 description. */
+  std::array<int, 12> map_joint_id_to_motor_board_id_;
```
Comment (MaximilienNaveau) [inline include/blmc_robots/solo8.hpp]: 12?

File: include/blmc_robots/solo8.hpp [hunk 5]
```diff
@@ -306,28 +307,31 @@ class Solo8
...
+
+  /** @brief Map the joint id to the motor port id, @see Solo12 description. */
+  std::array<int, 12> map_joint_id_to_motor_port_id_;
```
Comment (MaximilienNaveau) [inline include/blmc_robots/solo8.hpp]: 12? not 8?

File: src/solo8.cpp [hunk 6]
```diff
@@ -173,54 +130,34 @@ void Solo8::acquire_sensors()
...
   // acquire the slider positions
-  for (unsigned i=0 ; i<slider_positions_.size() ; ++i)
-  {
-    // acquire the slider
-    slider_positions_(i) = sliders_[i]->get_measurement()->newest_element();
-    // acquire the current contact states
-    contact_sensors_states_(i) =
-        contact_sensors_[i]->get_measurement()->newest_element();
-  }
+  // TODO: Acquire via Arduino.
```
Comment (MaximilienNaveau) [inline src/solo8.cpp]: Just a suggestion: /** @todo ... */ display the todo in the documentation. It is nice when you want to track the todo-list.

Comment (jviereck) [reply src/solo8.cpp]: Does `// @todo` also work?

File: include/blmc_robots/solo8.hpp (hunk 4 above)
Comment (jviereck) [reply include/blmc_robots/solo8.hpp]: Good catch :)

**Comment by jviereck** [comment]
@MaximilienNaveau thanks for your feedback!

I've added a first version of spi_joint_modules (haven't tried to compile or integrate it yet). Can you still take a look if you are okay with overall design direction?

  https://github.com/open-dynamic-robot-initiative/blmc_robots/pull/55/files#diff-61840fbfd01ef7f69da5dd59712afb65

A few comments:
* There is no abstraction for each joint module but the joint modules class handles each joint module directly
* There is no calibration functionality. This is to be added / done in a different way later (not sure what's the best way to tackle this yet)
* the joint modules class has a new `enable` and `is_ready` method. These act as helpers as there is no spi_bus instance etc around anymore.

**Comment by jviereck** [comment]
Hi @MaximilienNaveau ,

I tried to address all your changes. Still, GitHub says there are open requests of you. Can you take a look and maybe give approval on this PR?

**Pull request merged**